### PR TITLE
feat(text-model): support plain text

### PR DESCRIPTION
**Pull request by wittjosiah**

### CODE REVIEW

**Comment by wittjosiah** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2738** <a href="https://app.graphite.dev/github/pr/dxos/dxos/2738" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dxos/dxos/2738?utm_source=stack-comment).

**Comment by wittjosiah** [comment]
@thure something I did to the composer-app is causing this to happen 
image
if you have a moment to take a look

File: packages/apps/composer-app/src/proto/schema.proto [hunk 1]
```diff
@@ -13,5 +13,4 @@ message ComposerDocument {
```
Comment (richburdon) [inline packages/apps/composer-app/src/proto/schema.proto:13]: 1. we should now start to think about a types lib (similar to kai-types) where we have dxos types.
2. these types should not be app specific -- neither in namespace (dxos.app) -- nor name; i.e., this isn't a ComposerDocument; it's a Document that composer -- or kai -- uses. And would replace Document in kat-types.

File: packages/apps/patterns/react-composer/src/testing/replicator.ts [hunk 2]
```diff
@@ -175,11 +179,12 @@ export class Replicator {
 
 export type UseYjsModelOptions = {
   replicator: Replicator;
+  id: string;
   doc?: Doc;
 };
 
-export const usePlainYjsModel = ({ replicator, doc }: UseYjsModelOptions): PlainTextModel => {
-  const peer = useMemo(() => replicator.createPeer(doc), [doc]);
+export const useYjsModel = ({ replicator, id, doc }: UseYjsModelOptions): ComposerModel => {
```
Comment (richburdon) [inline packages/apps/patterns/react-composer/src/testing/replicator.ts:186]: Can we promote a more general useTextModel to react-client that can be parameterized by text type?

File: packages/apps/patterns/react-composer/src/components/RichText/RichText.stories.tsx [hunk 3]
```diff
@@ -3,58 +3,115 @@
...
-      />
-    </main>
-  );
+  }
 };
 
-export const Default = {
-  render: Story,
-  decorators: [ClientSpaceDecorator({ schema, count: 2 })]
+export const Echo = {
```
Comment (richburdon) [inline packages/apps/patterns/react-composer/src/components/RichText/RichText.stories.tsx:31]: Prefer WithEcho or some other storybook pattern/prefix.

File: packages/apps/patterns/react-composer/src/components/RichText/RichText.stories.tsx [hunk 4]
```diff
@@ -3,58 +3,115 @@
...
+      onCreateSpace: async (space) => {
+        const document = new ComposerDocument({ content: new Text('Hello, Storybook!', TextKind.RICH) });
+        await space?.db.add(document);
+      }
+    })
+  ]
+};
+
+const replicator = new Replicator(TextKind.RICH);
+export const Yjs = {
```
Comment (richburdon) [inline packages/apps/patterns/react-composer/src/components/RichText/RichText.stories.tsx:81]: Same naming issue

File: packages/apps/patterns/react-composer/src/components/RichText/RichText.stories.tsx [hunk 5]
```diff
@@ -3,58 +3,115 @@
...
+  render: (args: Omit<RichTextComposerProps, 'model'>) => {
+    const [generate, setGenerate] = useState(false);
+    const generateId = useId('generate');
+
+    const [id] = useState(PublicKey.random().toHex());
+    const model = useYjsModel({ id, replicator });
```
Comment (richburdon) [inline packages/apps/patterns/react-composer/src/components/RichText/RichText.stories.tsx:87]: Same point as above; better to have configurable hook.

File: packages/sdk/react-client/src/testing/useDataGenerator.tsx [hunk 6]
```diff
@@ -4,37 +4,37 @@
...
 
 type LoremOptions = {
-  plainText?: ReturnType<Doc['getText']>;
+  text?: YText | YXmlFragment;
   period?: number;
 };
 
-export const loremGenerator: DataGenerator<LoremOptions> = ({ plainText, period = 1000 }) => {
-  if (!plainText) {
+export const loremGenerator: DataGenerator<LoremOptions> = ({ text, period = 1000 }) => {
```
Comment (richburdon) [inline packages/sdk/react-client/src/testing/useDataGenerator.tsx:21]: textGenerator

**Comment by thure** [comment]
> @thure something I did to the composer-app is causing this to happen if you have a moment to take a look

Yep, it wasn't you though it was a fix we made for kai, I'll find a fix for Composer.

**Pull request merged**